diff --git a/net/core/dev.c b/net/core/dev.c
index 24631dd..d5675d1 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -176,6 +176,11 @@ static DEFINE_SPINLOCK(ptype_lock);
 static struct list_head ptype_base[PTYPE_HASH_SIZE] __read_mostly;
 static struct list_head ptype_all __read_mostly;	/* Taps */
 
+int (*skb_rx_accounter)(const struct sk_buff *skb);
+EXPORT_SYMBOL(skb_rx_accounter);
+int (*skb_tx_accounter)(const struct sk_buff *skb);
+EXPORT_SYMBOL(skb_tx_accounter);
+
 /*
  * The @dev_base_head list is protected by @dev_base_lock and the rtnl
  * semaphore.
@@ -1834,6 +1839,9 @@ int dev_hard_start_xmit(struct sk_buff *skb, struct net_device *dev,
 	int rc;
 	unsigned int skb_len;
 
+    if(skb_tx_accounter) 
+        (*skb_tx_accounter)(skb);
+
 	if (likely(!skb->next)) {
 		if (!list_empty(&ptype_all))
 			dev_queue_xmit_nit(skb, dev);
@@ -2758,10 +2766,14 @@ int __netif_receive_skb(struct sk_buff *skb)
 	if (!skb->tstamp.tv64)
 		net_timestamp(skb);
 
+
 	trace_netif_receive_skb(skb);
 	if (vlan_tx_tag_present(skb) && vlan_hwaccel_do_receive(skb))
 		return NET_RX_SUCCESS;
 
+    if(skb_rx_accounter) 
+        (*skb_rx_accounter)(skb);
+
 	/* if we've gotten here through NAPI, check netpoll */
 	if (netpoll_receive_skb(skb))
 		return NET_RX_DROP;
@@ -6461,6 +6473,8 @@ static int __init net_dev_init(void)
 	hotcpu_notifier(dev_cpu_callback, 0);
 	dst_init();
 	dev_mcast_init();
+    skb_rx_accounter = NULL;
+    skb_tx_accounter = NULL;
 	rc = 0;
 out:
 	return rc;
